en/middleware/micrortps.md

```diff
@@ -0,0 +1,344 @@
+  -c, --client          Flag for generate the client, by default is true if -a
+                        is not specified
+  -t MSGDIR, --topic-msg-dir MSGDIR
+                        Topics message dir, by default msg/
+  -o AGENTDIR, --agent-outdir AGENTDIR
+                        Agent output dir, by default
+                        src/modules/micrortps_bridge/micrortps_agent
+  -u CLIENTDIR, --client-outdir CLIENTDIR
+                        Client output dir, by default
+                        src/modules/micrortps_bridge/micrortps_client
+  -f FASTRTPSGEN, --fastrtpsgen-dir FASTRTPSGEN
+                        fastrtpsgen installation dir, by default /bin
+  --no-delete           Do not delete dir tree output dir(s)
+```
+
+- The argument `--send/-s` means that the application from PX4 side will send these messages, and the argument `--receive/-r` specifies which messages is going to be received.
+- The output appears in `CLIENTDIR` (`-o src/modules/micrortps_bridge/micrortps_client`, by default) and in the `AGENTDIR` (`-u src/modules/micrortps_bridge/micrortps_agent`, by default). 
+ 
+  > **Caution** This script erases the content of the `CLIENTDIR` and the `AGENTDIR` before creating new files and folders.
+
+- If no flag `-a` or `-c` is specified, both the client and the agent will be generated and installed.
+- The `-f` option may be needed if *Fast RTPS* was not installed in the default location (`-f /path/to/fastrtps/installation/bin`).
+
+An example of use:
+
+```sh
+$ cd /path/to/PX4/Firmware
+$ python Tools/generate_microRTPS_bridge.py -s msg/sensor_baro.msg -r msg/sensor_combined.msg
+```
+
+Checking the correct installation:
+
+```sh
+$ tree src/modules/micrortps_bridge/micrortps_agent
+src/modules/micrortps_bridge/micrortps_agent
+├── build
+├── CMakeLists.txt
+├── idl
+│   ├── sensor_baro_.idl
+│   └── sensor_combined_.idl
+├── microRTPS_agent.cxx
+├── microRTPS_transport.cxx
+├── microRTPS_transport.h
+├── RtpsTopics.cxx
+├── RtpsTopics.h
+├── sensor_baro_.cxx
+├── sensor_baro_.h
+├── sensor_baro_Publisher.cxx
+├── sensor_baro_Publisher.h
+├── sensor_baro_PubSubTypes.cxx
+├── sensor_baro_PubSubTypes.h
+├── sensor_combined_.cxx
+├── sensor_combined_.h
+├── sensor_combined_PubSubTypes.cxx
+├── sensor_combined_PubSubTypes.h
+├── sensor_combined_Subscriber.cxx
+└── sensor_combined_Subscriber.h
+ 2 directories, 20 files
+```
+
+```sh
+$ tree src/modules/micrortps_bridge/micrortps_client
+src/modules/micrortps_bridge/micrortps_client
+├── CMakeLists.txt
+├── microRTPS_client.cpp
+├── microRTPS_transport.cxx
+└── microRTPS_transport.h
+ 0 directories, 4 files
+```
+
+
+## PX4 Firmware: the micro RTPS client
+
+The agent application runs as both a uORB node and as a micro RTPS node. As a uORB node the agent could be subscribed to several topics as well as publish under internal uORB topics. The application receives from a internal publishers the messages, serializes the struct and writes it trough an UART
```

Comment:
What are you trying to say with "As a uORB node the agent could be subscribed to several topics as well as publish under internal uORB topics"

I think you're saying that it may be subscribed to internal uORB topics, and act as an external publisher over RTPS AND that it may be a publisher of uORB topics FROM remote uORB systems that it receives via RTPS. Correct?